### PR TITLE
[SPARK-49813] Publish Apache Spark 3.5.3 to docker registry

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

This PR proposes to publish Apache Spark 3.5.3 to docker registry

### Why are the changes needed?

This is one of required release step for Apache Spark

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

CI

### CODE REVIEW

**Comment by itholic** [comment]
cc @dongjoon-hyun would you mind taking a look when you find some time?

File: 3.5.3/scala2.12-java11-ubuntu/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    touch /opt/spark/RELEASE; \
+    chown -R spark:spark /opt/spark; \
+    echo "auth required pam_wheel.so use_uid" >> /etc/pam.d/su; \
+    rm -rf /var/lib/apt/lists/*
+
+# Install Apache Spark
+# https://downloads.apache.org/spark/KEYS
+ENV SPARK_TGZ_URL=https://archive.apache.org/dist/spark/spark-3.5.3/spark-3.5.3-bin-hadoop3.tgz \
+    SPARK_TGZ_ASC_URL=https://archive.apache.org/dist/spark/spark-3.5.3/spark-3.5.3-bin-hadoop3.tgz.asc \
+    GPG_KEY=None
```
Comment (Yikun) [inline 3.5.3/scala2.12-java11-ubuntu/Dockerfile:41]: The GPG key/value was wrong, so you need to rerun ./add-dockerfiles.sh 3.5.3 to regenerate dockerfiles after gpg kv fix.

Comment (itholic) [reply 3.5.3/scala2.12-java11-ubuntu/Dockerfile]: Oh just fixed it. Thanks for the comment!

**Comment by itholic** [comment]
@Yikun would mind checking CI why it's still failing after rerunning `./add-dockerfiles.sh 3.5.3` ??

**Comment by Yikun** [comment]
GPG validate Error, you can see https://github.com/apache/spark-docker/pull/63#issuecomment-2181836041 as reference.

**Comment by itholic** [comment]
Thanks @Yikun ! Just uploaded the public key

**Comment by itholic** [comment]
CI passed. Now can we just merge it?

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Review by Yikun — APPROVED** [review]
It would be better to add a jira number in PR title, otherwise LGTM

**Comment by itholic** [comment]
Thanks! Merged to master

**Comment by dongjoon-hyun** [comment]
For the record, I published the images.
- https://github.com/apache/spark-docker/actions/runs/11074454474

**Comment by itholic** [comment]
Got it. Thanks, @dongjoon-hyun !

**Pull request closed without merging**